templates/infra.json

```diff
@@ -8,340 +8,426 @@
-            "allowedValues": [
-                "node",
-                "dotnet",
-                "java",
-                "powershell"
-            ],
-            "metadata": {
-                "description": "The language worker runtime to load in the scaler function app."
+                "description": "GUID for engine contributor role on subscription."
             }
         },
-        "engineRuntime": {
+        "scalerContributorGUID": {
             "type": "string",
-            "defaultValue": "dotnet",
-            "allowedValues": [
-                "node",
-                "dotnet",
-                "java",
-                "powershell"
-            ],
+            "defaultValue": "[newGuid()]",
             "metadata": {
-                "description": "The language worker runtime to load in the engine function app."
+                "description": "GUID for scaler contributor role on subscription."
             }
         }
     },
     "variables": {
-        "functionAppName-scaler": "[concat(parameters('functName'), '-scaler')]",
-        "functionAppName-engine": "[concat(parameters('functName'), '-engine')]",
-        "hostingPlanName": "[concat(parameters('appName'), '-asp')]",
-        "applicationInsightsName": "[concat(parameters('appName'), '-appinsights')]",
-        "appConfigName": "[concat(parameters('appName'), '-appconfig')]",
-        "scalerFuncWorkerRuntime": "[parameters('scalerRuntime')]",
-        "engineFuncWorkerRuntime": "[parameters('engineRuntime')]",
-        "stgQueueName": "[concat(parameters('stgacctName'), '/default/', parameters('queueName'))]"
+        "contributorRole": "[subscriptionResourceId('Microsoft.Authorization/roleDefinitions', 'b24988ac-6180-42a0-ab88-20f7382dd24c')]"
     },
     "resources": [
         {
-            "type": "Microsoft.Storage/storageAccounts",
-            "apiVersion": "2019-06-01",
-            "name": "[parameters('stgacctName')]",
-            "location": "[parameters('location')]",
-            "sku": {
-                "name": "[parameters('storageAccountType')]"
-            },
-            "kind": "Storage",
-            "resources": [
-            ]
+            "type": "Microsoft.Resources/resourceGroups",
+            "apiVersion": "2020-06-01",
+            "name": "[concat(parameters('appName'), '-rg')]",
+            "location": "[deployment().location]",
+            "properties": {}
         },
         {
-            "type": "Microsoft.Storage/storageAccounts/queueServices",
-            "apiVersion": "2020-08-01-preview",
-            "name": "[concat(parameters('stgacctName'), '/default')]",
+            "type": "Microsoft.Authorization/roleAssignments",
+            "apiVersion": "2020-04-01-preview",
+            "name": "[parameters('engineContributorGUID')]",
             "dependsOn": [
-                "[resourceId('Microsoft.Storage/storageAccounts', parameters('stgacctName'))]"
+                "bellhopDeploy"
```

Comment:
Not important, but since this depends on bellhopDeploy, I would encourage you to put this below that object in the ARM template. Improves readability.